test/junit/scala/collection/MapTest.scala

```diff
@@ -128,4 +130,10 @@ class MapTest {
     assertFalse(Set("") == immutable.BitSet(1))
     assertFalse(Map("" -> 2) == scala.collection.immutable.LongMap(1L -> 2))
   }
+
+  @Test def `map is reachable from its keyset`: Unit =
+    assertFails(_.contains("held reference")) {
+      val m = Map("one" -> "eins")
+      assertNotReachable(m, m.keySet)(())
```

Comment:
maybe it should have a default body arg `(body: => Unit = ())`. I've seen this usage a couple of times this week.